src/openrct2/interface/Screenshot.cpp

```diff
@@ -93,98 +97,103 @@ static void screenshot_get_rendered_palette(rct_palette* palette)
     }
 }
 
-static int32_t screenshot_get_next_path(char* path, size_t size)
+static std::string screenshot_get_park_name()
 {
-    char screenshotPath[MAX_PATH];
+    char buffer[512];
+    format_string(buffer, sizeof(buffer), gParkName, &gParkNameArgs);
+    return buffer;
+}
 
+static opt::optional<std::string> screenshot_get_directory()
```

Comment:
I don't think the result should be optional, it should always return the full directory path even if it doesn't exist, the caller should make sure afterwards that the directory exists, this makes the code complex.